File: Parakh_24b0729/week-5/largest_oddnumber_string.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    string largestOddNumber(string num) {
        int end = -1;
        for (int i = num.size() - 1; i >= 0; i--) {
            // Check if the digit is odd
            if ((num[i] - '0') % 2 == 1) {
                end = i;
                break;  // no need to continue once last odd digit found
            }
        }
        if (end == -1) return "";  // no odd digit found
        return num.substr(0, end + 1);  // substring from start to end inclusive
    }
};

int main() {
    
    return 0;
}

File: Chavda_23B0368/Week_5/Largest_odd_number_string.java
package Week_5;

class Solution {
    public String largestOddNumber(String num) {
        for(int i = num.length()-1; i >= 0; i--) {
            char ch = num.charAt(i);
            if(ch=='1' || ch=='3' || ch=='5' || ch=='7' || ch=='9') {
                return num.substring(0, i+1);
            }
        }
        return "";
    }
}